spi/data-plane-selector/data-plane-selector-spi/src/main/java/org/eclipse/edc/connector/dataplane/selector/spi/instance/DataPlaneInstance.java

```diff
@@ -14,53 +14,172 @@
 
 package org.eclipse.edc.connector.dataplane.selector.spi.instance;
 
-import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import org.eclipse.edc.spi.EdcException;
 import org.eclipse.edc.spi.types.domain.DataAddress;
-import org.eclipse.edc.spi.types.domain.Polymorphic;
 
+import java.net.MalformedURLException;
 import java.net.URL;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
 
 /**
  * Representations of a data plane instance. Every DPF has an ID and a URL as well as a number, how often it was selected,
  * and a timestamp of its last selection time. In addition, there are extensible properties to hold specific properties.
  */
-@JsonSubTypes({
-        @JsonSubTypes.Type(value = DataPlaneInstanceImpl.class, name = "dataspaceconnector:dataplaneinstance")
-})
-public interface DataPlaneInstance extends Polymorphic {
+public class DataPlaneInstance {
 
-    /**
-     * Gets the unique identifier for this instance
-     */
-    String getId();
+    private Map<String, Object> properties;
+
+    @JsonProperty("allowedSourceTypes")
```

Comment:
this annotation is not necessary, as the JSON field is named the same as the property and the builder method.